### PR TITLE
feat: Develop Modal Components

### PR DESCRIPTION
**Pull request by juntaepark**

### CODE REVIEW

File: packages/web/src/components/swap/confirm-swap-modal/ConfirmSwapModal.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import { ComponentStory, ComponentMeta } from "@storybook/react";
+import { css, Theme } from "@emotion/react";
+import { action } from "@storybook/addon-actions";
+
+import ConfirmSwapModal from "./ConfirmSwapModal";
+import {
+  RecentdummyToken,
```
Comment (jinoosss) [inline packages/web/src/components/swap/confirm-swap-modal/ConfirmSwapModal.stories.tsx:8]: I think it's an unused variable.

File: packages/web/src/components/swap/confirm-swap-modal/ConfirmSwapModal.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    price: "$0.00",
+    gnosExchangePrice: "1250",
+    usdExchangePrice: "($1541.55)",
+    balance: "0",
+    tokenLogo:
+      "https://raw.githubusercontent.com/Uniswap/assets/master/blockchains/ethereum/assets/0x2b591e99afE9f32eAA6214f7B7629768c40Eeb39/logo.png",
+  },
+};
+
+const wrapper = (theme: Theme) => css`
```
Comment (jinoosss) [inline packages/web/src/components/swap/confirm-swap-modal/ConfirmSwapModal.stories.tsx:62]: I get a lint error with an unused variable, is it necessary?

File: packages/web/src/components/swap/select-token-modal/SelectTokenModal.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { ComponentStory, ComponentMeta } from "@storybook/react";
+import { css, Theme } from "@emotion/react";
+import { action } from "@storybook/addon-actions";
+
+import SelectTokenModal from "./SelectTokenModal";
+import {
+  RecentdummyToken,
```
Comment (jinoosss) [inline packages/web/src/components/swap/select-token-modal/SelectTokenModal.stories.tsx:8]: I think it's an unused variable.

File: packages/web/src/components/swap/select-token-modal/SelectTokenModal.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import SearchMenuModal from "./SelectTokenModal";
+import { render } from "@testing-library/react";
+import { Provider as JotaiProvider } from "jotai";
+import GnoswapThemeProvider from "@providers/gnoswap-theme-provider/GnoswapThemeProvider";
+import {
+  RecentdummyToken,
```
Comment (jinoosss) [inline packages/web/src/components/swap/select-token-modal/SelectTokenModal.spec.tsx:6]: I think it's an unused variable.

File: packages/web/src/components/swap/select-token-modal/SelectTokenModal.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+} from "@containers/swap-container/SwapContainer";
+
+describe("SearchMenuModal Component", () => {
+  it("should render", () => {
+    const mockProps = {
+      search: () => null,
+      keyword: "",
+      onSelectTokenModal: () => null,
+      coinList: coinList,
+      changeToken: (token: coinListProps, type: string) => null,
```
Comment (jinoosss) [inline packages/web/src/components/swap/select-token-modal/SelectTokenModal.spec.tsx:21]: The test succeeds even without the argument.

File: packages/web/src/components/swap/select-token-modal/SelectTokenModal.styles.ts [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+        }
+        .coin-logo {
+          width: 24px;
+          height: 24px;
+        }
+      }
+    }
+  }
+
+  .list-wrap {
```
Comment (jinoosss) [inline packages/web/src/components/swap/select-token-modal/SelectTokenModal.styles.ts:149]: I think `list-wrap` should be scrollable if the list grows.

File: packages/web/src/components/swap/select-token-modal/SelectTokenModal.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+            <span>Select a token</span>
+            <div className="close-wrap" onClick={onSelectTokenModal}>
+              <IconClose className="close-icon" />
+            </div>
+          </div>
+          <div className="search-wrap">
+            <InputStyle
+              placeholder={"Search name or paste address"}
+              value={keyword}
+              onChange={search}
```
Comment (jinoosss) [inline packages/web/src/components/swap/select-token-modal/SelectTokenModal.tsx:60]: It seems to be missing the ability to retrieve a list of tokens by the `search` function.

**Comment by jinoosss** [comment]
LGTM !
I left a few comments.

File: packages/web/src/components/swap/confirm-swap-modal/ConfirmSwapModal.stories.tsx (hunk 1 above)
Comment (juntaepark) [reply packages/web/src/components/swap/confirm-swap-modal/ConfirmSwapModal.stories.tsx:8]: Deleted.

File: packages/web/src/components/swap/confirm-swap-modal/ConfirmSwapModal.stories.tsx (hunk 2 above)
Comment (juntaepark) [reply packages/web/src/components/swap/confirm-swap-modal/ConfirmSwapModal.stories.tsx:62]: I do not need.

File: packages/web/src/components/swap/select-token-modal/SelectTokenModal.stories.tsx (hunk 3 above)
Comment (juntaepark) [reply packages/web/src/components/swap/select-token-modal/SelectTokenModal.stories.tsx:8]: deleted !

File: packages/web/src/components/swap/select-token-modal/SelectTokenModal.spec.tsx (hunk 4 above)
Comment (juntaepark) [reply packages/web/src/components/swap/select-token-modal/SelectTokenModal.spec.tsx:6]: deleted !

File: packages/web/src/components/swap/select-token-modal/SelectTokenModal.spec.tsx (hunk 5 above)
Comment (juntaepark) [reply packages/web/src/components/swap/select-token-modal/SelectTokenModal.spec.tsx:21]: deleted !

File: packages/web/src/components/swap/select-token-modal/SelectTokenModal.styles.ts (hunk 6 above)
Comment (juntaepark) [reply packages/web/src/components/swap/select-token-modal/SelectTokenModal.styles.ts:149]: Changed to enable scrolling.

File: packages/web/src/components/swap/select-token-modal/SelectTokenModal.tsx (hunk 7 above)
Comment (juntaepark) [reply packages/web/src/components/swap/select-token-modal/SelectTokenModal.tsx:60]: good ! added.

**Review by jinoosss — APPROVED** [review]
Approve  !

**Pull request merged**